doc/src/gp-details.rst

```diff
@@ -176,3 +176,85 @@ would:
 * Evaluate the prior covariance :math:`K_{**}`
 * Compute the posterior covariance :math:`K_{**} - Q_{f*}^T Q_{f*}`
 
+++++++++++++++++++++++++++++++++++++++++
+Cross Validation Implementation Details
+++++++++++++++++++++++++++++++++++++++++
+
+In `Gaussian Processes for Machine Learning`_ (Section 5.4.2) they describe an efficient way for making leave one out predictions, here we expand that same trick to enable making leave one group out predictions.
+
+Consider the case where we have a set of observations, :math:`y`, and we would like to make leave one group out cross validated predictions and by groups we mean independent sets of one or more variables.
+
+We start with our GP,
+
+.. math::
+
+    \mathbf{f} \sim \mathcal{N}\left(0, \Sigma \right)
+
+Which we can then break into groups,
+
+.. math::
+
+    \begin{bmatrix} \mathbf{\hat{y}} \\ \mathbf{y_i} \end{bmatrix} \sim \mathcal{N}\left(\begin{bmatrix} 0 \\ 0 \end{bmatrix}, \begin{bmatrix}\hat{\Sigma} & S \\ S^T & C \end{bmatrix}\right)
+
+Where we will be using a subset of observations, :math:`\hat{y}` to make predictions for a held out set of locations, :math:`x_i`.  We can do this directly using the Gaussian process predict formula,
+
+.. math::
+
+    [\mathbf{y_i}|\mathbf{\hat{y}}=\hat{y}] \sim \mathcal{N}\left(S^T \hat{\Sigma}^{-1} \hat{y}, C - S^T \hat{\Sigma}^{-1} S\right)
+
+But doing so would require computing :math:`\hat{\Sigma}^{-1}` for every group, :math:`i`, that we hold out.  So if we're doing leave one out with :math:`n` observations we have to do the :math:`\mathcal{O}(n^3)` inversion :math:`n` times leading to :math:`\mathcal{O}(n^4)` complexity which will quickly get infeasible.
+
+However, in the process of fitting our GP we'll need to end up computing the inverse of the full covariance, :math:`\Sigma^{-1}` as well as what we've been calling the information vector, :math:`v = \Sigma^{-1} y`.  By using block inversion we get,
+
+.. math::
+
+    \Sigma^{-1} = \begin{bmatrix}
+    \left(\hat{\Sigma} - S C^{-1} S^T\right)^{-1} & -\left(\hat{\Sigma} - S C^{-1} S^T\right)^{-1}SC^{-1} \\
```

Comment:
I think there might be some typos in this formula.  We should never see `\hat{\Sigma} - S C^{-1} S^T` as the middle term of any of these blocks -- I think `\Sigma` and `C` are swapped here.

Reply:
Also it looks like we are post-multiplying by `S C^{-1}` rather than pre-multiplying by `S \Sigma^{-1}`.

Reply:
There are two different equivalent forms for the block inversion (you either form schur complements over the top left or bottom right block), https://en.wikipedia.org/wiki/Block_matrix#Block_matrix_inversion. I was picking and choosing between them to get terms which cancel easier.

Reply:
Thanks!  I didn't think about this.